### DISCUSSION TITLE
Metadata in (%flac(%ogg) streams?

### DISCUSSION THREAD

**Discussion by privacyguy123** [savonet/liquidsoap#2309] [User Support]
I have searched high and low but truth be told I'm not so good at this stuff ... can't get FLAC streams to send metadata - what am I doing wrong?

**Comment by toots** [comment]
Most likely a limitation of icecast. The ogg container has its metadata placed at the beginning of the stream. For some ogg streams, icecast knows how to replay the header packet when a new listener connects but for flac it might not.

This means that new listeners won't get initial metadata but will get them the next time there's a new metadata packet.

You can verify that liquidsoap does indeed encode metadata in `%ogg(%flac)` streams by saving them to a file:
```ruby
output.file(fallible=true, %ogg(%flac), "/tmp/bla.ogg", playlist("/tmp/pl"))
```
And then:
```shell
% ffprobe /tmp/bla.ogg
```

**Comment by privacyguy123** [comment]
I've ended up even more confused than when I started - so ```icy_metadata``` being on or off is irrelevant or? I notice on my Icecast admin panel it will straight up refuse metadata manual update saying it can't be done from URL and there is definitely none being sent - I can confirm this by changing output encoder to MP3 and trying a manual update, still no automatic metadata though. Is there a chance Tuneblade for Windows just isn't sending any over to Shairport-sync?

Should that code snippet be at the bottom of my liquidsoap.liq file underneath ```output.icecast ...``` ?

**Comment by Vaalyn** [comment]
Neither Icecast nor Icecast-KH support metadata for the `ogg/flac` format.

See this 12 year old related issue for Icecast: https://gitlab.xiph.org/xiph/icecast-server/-/issues/1655
And these issues for Icecast-KH: https://github.com/karlheyes/icecast-kh/issues/153 & https://github.com/karlheyes/icecast-kh/issues/324

They won't display any metadata on the Icecast pages and the Icecast API but if you have `icy_metadata=false` the audio stream itself will contain metadata so players like VLC should show that.

Small Edit: Looks like Icecast itself just fixed this 1 week ago after 12 years by making use of the vorbis comments. Very new and probably not yet available in a fully released Icecast version.

**Reply by privacyguy123** [reply]
So what you're saying is - expect an Icecast update?

**Reply by Vaalyn** [reply]
Yes, probably.
As far as I can see the patch that is mentioned in the Icecast issue I linked above is only merged into a development branch which means it's not even available in a beta version atm. No idea how long it will take until they add it to the 2.5 beta or until they release 2.5 fully.

**Reply by privacyguy123** [reply]
Even with this update ... is there some convoluted process of config editing I'm missing? I couldn't get my mp3 stream to automatically supply metadata either.
